WordPress/gutenberg
# Editor: Consider single unmodified default block as empty content

Related: #9626

This pull request seeks to consider a post which consists only of a single unmodified default block (paragraph) to be empty, rather than serialize as currently:

```

<p></p>

```

This will be used toward #9626 in preventing a user from deleting all of blocks to an entirely empty post while ensuring that empty post is saved as truly empty.

**Implementation notes:**

Since the behavior here is similar in form to a previously-added `removep` logic (#8077), it refactors the approach to consider block saveability by filter extension. This also helps support the need here for preserving existing logic of "is saveable", where the omission of the unmodified default block can be a factor in consideration. It introduces two new filterable selectors `getBlocksForSave` and `getBlockContent`.

**Testing instructions:**

Verify that a post is not saveable once clicking into the writing prompt.

Verify that a post's content is correctly saved once entered.

Verify that if a title is assigned but content exists only as the unmodified default block, that the saved content is empty.

Verify that `removep` is applied to the unknown block type (freeform).
